### PR TITLE
[Security Content] Add Investigation Guides for ML rules

### PR DESCRIPTION
**Pull request by w0rk3r**
## Issues

https://github.com/elastic/security-team/issues/4703

## Summary

Adds Investigation Guides to the following Rules:

- Spike in Logon Events from a Source IP **(Renamed to)** Spike in Successful Logon Events from a Source IP
- Spike in Failed Logon Events
- Rare User Logon
- Unusual Hour for a User to Logon
- Unusual Process For a Linux Host
- Anomalous Windows Process Creation
- Anomalous Process For a Linux Population
- Spike in Network Traffic To a Country
- Anomalous Process For a Windows Population
- Unusual Process For a Windows Host

### CODE REVIEW

File: rules/ml/credential_access_ml_auth_spike_in_logon_events.toml [hunk 1]
```diff
@@ -22,7 +22,46 @@ from = "now-30m"
 interval = "15m"
 license = "Elastic License v2"
 machine_learning_job_id = "auth_high_count_logon_events"
-name = "Spike in Logon Events"
+name = "Spike in Failed Logon Events"
```
[the remark this answers is not in the document]
Comment (w0rk3r) [reply rules/ml/credential_access_ml_auth_spike_in_logon_events.toml:25]: Nice catch! I mixed up those for sure

File: rules/ml/persistence_ml_rare_process_by_host_windows.toml [hunk 2]
```diff
@@ -26,38 +26,83 @@ machine_learning_job_id = ["v3_rare_process_by_host_windows"]
...
+      - !{osquery{"query":"SELECT description, display_name, name, path, pid, service_type, start_type, status, user_account FROM services","label":"Osquery - Retrieve All Services"}}
+      - !{osquery{"query":"SELECT description, display_name, name, path, pid, service_type, start_type, status, user_account FROM services WHERE NOT (user_account LIKE \"%LocalSystem\" OR user_account LIKE \"%LocalService\" OR user_account LIKE \"%NetworkService\" OR user_account == null)","label":"Osquery - Retrieve Services Running on User Accounts"}}
+      - !{osquery{"query":"SELECT concat('https://www.virustotal.com/gui/file/', sha1) AS VtLink, name, description, start_type, status, pid, services.path FROM services JOIN authenticode ON services.path = authenticode.path OR services.module_path = authenticode.path JOIN hash ON services.path = hash.path WHERE authenticode.result != \"trusted\"","label":"Retrieve Service Unisgned Executables with Virustotal Link"}}
+  - Retrieve the files' SHA-256 hash values using the PowerShell `Get-FileHash` cmdlet and search for the existence and
+  reputation of the hashes in resources like VirusTotal, Hybrid-Analysis, CISCO Talos, Any.run, etc.
```
Comment (joepeeples) [inline rules/ml/persistence_ml_rare_process_by_host_windows.toml:72]: The nested list levels here are pretty complex and hard to follow. Would it be possible to have just two or three levels at most? It seems like most of the actions would OK on the same level:

- Examine the host for artifacts:
  - Analyze the process executable
  - Observe and collect
  - Use the ED network events
  - Examine the DNS cache
    - osquery example
  - Use the ED registry events
  - Examine the host services
    - osquery example
    - osquery example
    - osquery example
  - Retrieve the files' SHA-256

File: rules/ml/persistence_ml_windows_anomalous_process_all_hosts.toml [hunk 3]
```diff
@@ -26,14 +26,84 @@ machine_learning_job_id = ["v3_windows_anomalous_process_all_hosts"]
...
+      - !{osquery{"query":"SELECT description, display_name, name, path, pid, service_type, start_type, status, user_account FROM services","label":"Osquery - Retrieve All Services"}}
+      - !{osquery{"query":"SELECT description, display_name, name, path, pid, service_type, start_type, status, user_account FROM services WHERE NOT (user_account LIKE \"%LocalSystem\" OR user_account LIKE \"%LocalService\" OR user_account LIKE \"%NetworkService\" OR user_account == null)","label":"Osquery - Retrieve Services Running on User Accounts"}}
+      - !{osquery{"query":"SELECT concat('https://www.virustotal.com/gui/file/', sha1) AS VtLink, name, description, start_type, status, pid, services.path FROM services JOIN authenticode ON services.path = authenticode.path OR services.module_path = authenticode.path JOIN hash ON services.path = hash.path WHERE authenticode.result != \"trusted\"","label":"Retrieve Service Unisgned Executables with Virustotal Link"}}
+  - Retrieve the files' SHA-256 hash values using the PowerShell `Get-FileHash` cmdlet and search for the existence and
+  reputation of the hashes in resources like VirusTotal, Hybrid-Analysis, CISCO Talos, Any.run, etc.
```
Comment (joepeeples) [inline rules/ml/persistence_ml_windows_anomalous_process_all_hosts.toml:72]: See earlier comment about nested list levels.

File: rules/ml/persistence_ml_windows_anomalous_process_creation.toml [hunk 4]
```diff
@@ -27,6 +27,86 @@ interval = "15m"
...
+      - !{osquery{"query":"SELECT description, display_name, name, path, pid, service_type, start_type, status, user_account FROM services","label":"Osquery - Retrieve All Services"}}
+      - !{osquery{"query":"SELECT description, display_name, name, path, pid, service_type, start_type, status, user_account FROM services WHERE NOT (user_account LIKE \"%LocalSystem\" OR user_account LIKE \"%LocalService\" OR user_account LIKE \"%NetworkService\" OR user_account == null)","label":"Osquery - Retrieve Services Running on User Accounts"}}
+      - !{osquery{"query":"SELECT concat('https://www.virustotal.com/gui/file/', sha1) AS VtLink, name, description, start_type, status, pid, services.path FROM services JOIN authenticode ON services.path = authenticode.path OR services.module_path = authenticode.path JOIN hash ON services.path = hash.path WHERE authenticode.result != \"trusted\"","label":"Retrieve Service Unisgned Executables with Virustotal Link"}}
+  - Retrieve the files' SHA-256 hash values using the PowerShell `Get-FileHash` cmdlet and search for the existence and
+  reputation of the hashes in resources like VirusTotal, Hybrid-Analysis, CISCO Talos, Any.run, etc.
```
Comment (joepeeples) [inline rules/ml/persistence_ml_windows_anomalous_process_creation.toml:75]: See earlier comment about nested list levels.

**Review by joepeeples — COMMENTED** [review]
Solid content, thanks for preparing it! Added a few repeated edit suggestions.

**Review by imays11 — APPROVED** [review]
Great job!

File: rules/ml/persistence_ml_rare_process_by_host_linux.toml [hunk 5]
```diff
@@ -26,11 +26,54 @@ machine_learning_job_id = ["v3_rare_process_by_host_linux"]
...
-- Examine the history of execution. If this process only manifested recently, it might be part of a new software package. If it has a consistent cadence (for example if it runs monthly or quarterly), it might be part of a monthly or quarterly business process.
-- Examine the process arguments, title and working directory. These may provide indications as to the source of the program or the nature of the tasks it is performing."""
+###  Investigating Unusual Process For a Linux Host
+
+Searching for abnormal Linux processes is a good methodology to find potentially malicious activity within a network.
+Understanding what is commonly run within an environment and developing baselines for legitimate activity can help
+uncover potential malware and suspicious behaviors.
+
+This rule uses a machine learning job to detect a Linux process that is rare and unusual for an individual Linux host in your fleet.
+#### Possible investigation steps
```
[the remark this answers is not in the document]
Comment (nastasha-solomon) [reply rules/ml/persistence_ml_rare_process_by_host_linux.toml:36]: Quick question about this sentence - in this instance, is this the Elastic Fleet or just a general group (i.e., a fleet of hosts). If it's the former, I'd capitalize the "f" in fleet. If it's the latter, it might be worth choosing a different term since "Fleet" is used in a very specific context within Elastic products. For example:

```suggestion
This rule uses a machine learning job to detect a Linux process that is rare and unusual for an individual Linux host within your group of hosts.

#### Possible investigation steps
```

**Review by nastasha-solomon — APPROVED** [review]
Just had the one suggestion. Otherwise, LGTM! Thanks again for putting this together :)

File: rules/ml/persistence_ml_rare_process_by_host_linux.toml (hunk 5 above)
[the remark this answers is not in the document]
Comment (w0rk3r) [reply rules/ml/persistence_ml_rare_process_by_host_linux.toml:36]: Replaced it with `environment`.

File: rules/ml/persistence_ml_rare_process_by_host_windows.toml (hunk 2 above)
Comment (w0rk3r) [reply rules/ml/persistence_ml_rare_process_by_host_windows.toml:72]: I'll keep this noted, but as those are implemented in other Rules that are not involved in this PR, so I'll not change the format by now

**Review by Samirbous — APPROVED** [review]
